src/components/authentication/authentication.resolver.ts

```diff
@@ -13,14 +16,31 @@ import { AuthenticationService } from './authentication.service';
 export class AuthenticationResolver {
   constructor(
     private readonly authService: AuthenticationService,
-    private readonly userService: UserService
+    private readonly userService: UserService,
+    private readonly config: ConfigService
   ) {}
 
   @Mutation(() => CreateSessionOutput, {
     description: 'Create a session',
   })
-  async createSession(): Promise<CreateSessionOutput> {
+  async createSession(
+    @Args('browser') browser: boolean,
+    @Context('response') res: Response
+  ): Promise<CreateSessionOutput> {
     const token = await this.authService.createToken();
+
+    if (browser) {
+      // http cookies must have an expiration in order to persist, so we're setting it to 10 years in the future
+      const expires = DateTime.local().plus({ years: 10 }).toJSDate();
+
+      res.cookie(this.config.sessionCookieName, token, {
+        expires,
+        httpOnly: true,
+        path: '/',
+        domain: this.config.sessionCookieDomain,
+      });
+    }
```

Comment:
Needs return so token is excluded

Reply:
Hmm, one thing I didn't anticipate here is we can't blindly return in this resolver. Because the `CreateSessionOutput` is a required field in graphql. What do you think the best way to go about handling this is?

Reply:
Yeah we should change that DTO to make that property optional and a description saying it's returned if browser is false. A description on the browser arg would be good too.

Reply:
Sounds good 👍